### DISCUSSION TITLE
MQTT Publish log-level

### DISCUSSION THREAD

**Discussion by Dirnei** [Koenkk/zigbee2mqtt#5633] [General]
Hi!

I really like the new logging topic but I wonder if I am the only one that thinks the `MQTT publish` log is really a info log. I would rather say that it is debug.

![image](https://user-images.githubusercontent.com/912432/104028505-13f48380-51c9-11eb-953f-5c0ad10ecdda.png)

IMHO it produces to much noise, because it is always followed by the other message that it publishes right afterwards. 

I'm also curious what other messages will be published under info.

**Comment by Koenkk** [comment]
Can you provide your debug log? I don't expect message send to `zigbee2mqtt/bridge/logging` to be logged with a `MQTT publish..` (at least not on z2m 1.17.0

**Reply by Dirnei** [reply]
I haven't configured debug logging. My configuration is mostly default. If you want I can enable it. But here are screenshots from the current logfile (only info logs) and some messages captured with a MQTT client. As you see there is always a info log that says that it will publish a message. And yes I'm on 1.17.0
![image](https://user-images.githubusercontent.com/912432/104068084-9cdce080-5204-11eb-92e7-b86a1b2841de.png)
![image](https://user-images.githubusercontent.com/912432/104067811-22ac5c00-5204-11eb-8de9-388935834b07.png)

**Reply by Koenkk** [reply]
Ah yes, that's indeed expected. In my opinion it's handy for most users to know what is published actually. But let's see if anybody reacts on this topic.

**Reply by FreakErn** [reply]
<del>How can i disable these Messages? Even setting log_level will not change anything.
I Updated a couple of days ago to 1.16.2 and now my Loxone Miniserver with 400Mhz single core has to parse all these nonsense MQTT messages.</del>

<del>If all devices would have been in zigbee2mqtt/devices/... and the logging would be in zigbee2mqtt/logging i would just subscribe to the other topic but the way it is right now is unusable for me and i have to revert my 1.16.2 update...</del>

/Edit
I just commented out the log message in lib/mqtt.js at line 128

**Reply by JJanssen123** [reply]
I agree with TS. I don't need all normal messages that are sent to/from devices to be present in the  logging topic as well. But I do like to see all the other info messages there. When I want to see the normal messages I'll subscribe to their channels.

**Reply by ivanfmartinez** [reply]
An easy way to disable this lot of messages will be very good. 
My log is filled with this publish info messages, and that messages can read directly on the mqtt
I use the option attribute_and_json which publishes more messages....

**Comment by FreakErn** [comment]
Could we get a Configuration property to disable these information Messages? 
These are useless for me and produce a lot of overhead.

**Comment by DrTron** [comment]
Interesting. I'm also on 1.17.0 and I do not have a zigbee2mqtt/bridge/logging topic....

**Reply by timoline** [reply]
I think you've got a good point here

**Comment by dyfrgi** [comment]
I agree - this log message is far too spammy. I don't need several log messages per minute, per lightswitch, stored in my system logs. It looks like this can also be disabled by changing the log level of the z2m:mqtt namespace, as documented: https://www.zigbee2mqtt.io/guide/configuration/logging.html#reducing-mqtt-traffic. From the code it looks like that's the only message in this namespace at info level.
